src/test/java/com/training/api/tests/cat/CatApiTests.java

```diff
@@ -1,79 +1,91 @@
-        int index = breedName.indexOf(name);
-        String id = response.jsonPath().getString(String.format("id[%s]",index));
-        logger.info(id);
-        Assertions.assertEquals(breedId, id,"Id incorrect");
+        Assertions.assertEquals(id, breedId, "Id incorrect");
     }
 
     @ParameterizedTest
     @CsvFileSource(resources = "/getWikipediaUrl.csv")
     public void testGetWikipediaUrl(String id, String url) {
+        io.restassured.response.Response response = new RestEngine()
+                .getResponse(String.format("/v1/images/search?breed_ids=%s", id), headerMap);
+
+        Assertions.assertEquals(200, response.getStatusCode(), "Request Unsuccessful");
 
-        String path = String.format("/v1/images/search?breed_ids=%s", id);
-        Response response = httpRequest.when().get(path);
-        Assertions.assertEquals( 200,response.getStatusCode(),"Request Unsuccessful");
+        List<com.training.pojos.cat.search.Response> responseList =
+                Arrays.asList(response.as(com.training.pojos.cat.search.Response[].class));
 
-        String wikipediaUrl = response.jsonPath().getString("breeds[0].wikipedia_url");
-        Assertions.assertTrue(wikipediaUrl.contains(url), "Url is not correct");
+
+        Assertions.assertEquals(url,
+                responseList.get(0).getBreeds().get(0).getWikipedia_url(),
+                "Wikipedia Url does not match");
     }
 
     @Test
     public void testPostVote() {
 
-        JSONObject vote = new JSONObject();
-        vote.put("image_id", "asf2");
-        vote.put("sub_id", "test05042021-9");
-        vote.put("value", 1);
-
-        Response response = httpRequest
-                .contentType("application/json")
-                .body(vote.toJSONString())
-                .when()
-                .post("/v1/votes");
-
-        Assertions.assertEquals(200,response.getStatusCode(), "Vote not posted successfully");
-
-        int id = response.jsonPath().getInt("id");
-        response = httpRequest.get("/v1/votes");
-        List<Integer> list = response.jsonPath().getList("id");
-
-        Assertions.assertTrue(list.contains(id),"Vote was not posted successfully");
+        Request request = new Request();
+        request.setImage_id("asf2");
+        request.setSub_id("test09042021-1");
+        request.setValue(1);
+
+        io.restassured.response.Response response = new RestEngine().getResponse("/v1/votes", headerMap,
+                new Gson().toJson(request));
+
+        Assertions.assertEquals(200, response.getStatusCode(),
+                "Vote not posted successfully");
+
+        response = new RestEngine().getResponse("/v1/votes", headerMap);
+
+        boolean idPresent = false;
+        for (Response responseElement :
+                Arrays.asList(response.as(com.training.pojos.cat.vote.Response[].class))) {
+            if (responseElement.getId().equals
+                    (response.as(com.training.pojos.cat.vote.Response.class).getId())) {
```

Comment:
I haven't told you to call direclty response.as(com.training.pojos.cat.vote.Response.class).getId() in if ,
call that outside for loop and define it in variable and use that variable in if(), 
Bcoz you are calling getId on every iteration and the value of getID will be always the same for every iteration.
In this case, if you call getID on every iteration then the performance of testPostVote test case will get slow